### PR TITLE
Make Search() and RangeSearch() properly handle inner exceptions

### PR DESCRIPTION
**Pull request by alexanderguzhva**
related issue: #149 
/kind improvement

### CODE REVIEW

File: src/index/flat/flat.cc [hunk 1]
```diff
@@ -113,9 +113,15 @@ class FlatIndexNode : public IndexNode {
...
             }
+            // wait for the completion
             for (auto& fut : futs) {
                 fut.wait();
             }
+            // check for exceptions. value() is {}, so either
+            //   a call does nothing, or it throws an inner exception.
+            for (auto& fut : futs) {
+                fut.result().value();
+            }
```
Comment (liliu-z) [inline src/index/flat/flat.cc:124]: Good catch!

**Review by liliu-z — COMMENTED** [review]
/lgtm

**Review by chasingegg — APPROVED** [review]

**Pull request merged**